core/domain/exp_domain.py

```diff
@@ -2430,9 +2450,177 @@ def _convert_states_v33_dict_to_v34_dict(cls, states_dict):
         for key, state_dict in states_dict.items():
             states_dict[key] = state_domain.State.convert_html_fields_in_state(
                 state_dict,
-                html_validation_service.add_math_content_to_math_rte_components)
+                html_validation_service.add_math_content_to_math_rte_components,
+                state_uses_old_interaction_cust_args_schema=True)
+
         return states_dict
 
+    @classmethod
+    def _convert_states_v34_dict_to_v35_dict(cls, states_dict):
+        """Converts from version 34 to 35. Version 35 adds translation support
+        for interaction customization arguments. This migration converts
+        customization arguments whose schemas have been changed from unicode to
+        SubtitledUnicode or html to SubtitledHtml. It also populates missing
+        customization argument keys on all interactions, removes extra
+        customization arguments, normalizes customization arguments against
+        its schema, and changes PencilCodeEditor's customization argument
+        name from initial_code to initialCode.
+
+        Args:
+            states_dict: dict. A dict where each key-value pair represents,
+                respectively, a state name and a dict used to initialize a
+                State domain object.
+
+        Returns:
+            dict. The converted states_dict.
+        """
+        for state_dict in states_dict.values():
+            # Find maximum existing content_id index.
+            max_existing_content_id_index = -1
+            translations_mapping = state_dict[
+                'written_translations']['translations_mapping']
+            for content_id in translations_mapping:
+                content_id_suffix = content_id.split('_')[-1]
+                if content_id_suffix.isdigit():
+                    max_existing_content_id_index = max(
+                        max_existing_content_id_index,
+                        int(content_id_suffix)
+                    )
+                for lang_code in translations_mapping[content_id]:
+                    translations_mapping[
+                        content_id][lang_code]['data_format'] = 'html'
+                    translations_mapping[
+                        content_id][lang_code]['translation'] = (
+                            translations_mapping[content_id][lang_code]['html'])
+                    del translations_mapping[content_id][lang_code]['html']
+
+            next_content_id_index = max_existing_content_id_index + 1
+
+            all_new_content_ids = []
+            interaction_id = state_dict['interaction']['id']
+            if interaction_id is not None:
+                ca_dict = state_dict['interaction']['customization_args']
```

Comment:
> Since customization_args is coming from a state_dict, instead of State, shouldn't we assume that it is a dict? If not, why do we make the same assumption for other elements of the state that are also dicts? For example for state_dict['interaction']['hints], we don't have to name field hints as hints_dicts, even though it is the dict representation of Hints domain object.

Good question -- it should be hint_dicts too, actually.